app/src/components/RequestTable.js

```diff
@@ -22,65 +26,172 @@ function RequestTable({ requests, token, onSubmit, onWithdraw }) {
-          entries={requests
-            .sort(({ date: dateLeft }, { date: dateRight }) =>
-              // Sort by date descending
-              compareDesc(dateLeft, dateRight)
-            )
-            .map(r => [
-              r.requestId,
-              r.date,
-              r.depositAmount,
-              r.depositSymbol,
-              r.depositToken,
-              r.depositName,
-              r.depositDecimals,
-              r.requestAmount,
-              r.status,
-              token.symbol,
-              token.decimals,
-            ])}
-          renderEntry={([
-            requestId,
-            date,
-            depositAmount,
-            depositSymbol,
-            depositTokenAddress,
-            depositName,
-            depositDecimals,
-            requestedAmount,
-            status,
-            requestedSymbol,
-            requestedDecimals,
-          ]) => [
-            <time>{format(date, 'dd/MM/yy')}</time>,
-            <Text>{`${formatTokenAmountSymbol(depositSymbol, depositAmount, false, depositDecimals)} `}</Text>,
-            <Text>{`${formatTokenAmountSymbol(requestedSymbol, requestedAmount, false, requestedDecimals)} `}</Text>,
-            <Status status={status}>{`${status}`}</Status>,
-            <ContextMenu disabled={status != requestStatus.PENDING}>
-              {status === requestStatus.PENDING && (
-                <ContextMenuItem onClick={() => handleSubmit(requestId)}>
-                  <IconWrapper>
-                    <IconVote />
-                  </IconWrapper>
-                  <div css="margin-left: 15px">Submit</div>
-                </ContextMenuItem>
-              )}
-              {status === requestStatus.PENDING && (
-                <ContextMenuItem onClick={() => handleWithdraw(requestId)}>
-                  <IconWrapper>
-                    <IconCoin />
-                  </IconWrapper>
-                  <div css="margin-left: 15px">Withdraw</div>
-                </ContextMenuItem>
-              )}
-            </ContextMenu>,
-          ]}
+          fields={
+            ownRequests
+              ? ['Request Date', 'Deposited', 'Requested', 'Status', 'Actions']
+              : ['Request Date', 'Requester', 'Deposited', 'Requested', 'Status', 'Actions']
+          }
+          entries={getEntries}
+          renderEntry={
+            !ownRequests
+              ? ([
+                  requestId,
+                  date,
+                  requesterAddress,
+                  depositAmount,
+                  depositSymbol,
+                  depositTokenAddress,
+                  depositName,
+                  depositDecimals,
+                  requestedAmount,
+                  status,
+                  requestedSymbol,
+                  requestedDecimals,
+                ]) =>
+                  getRow(
+                    requestId,
+                    date,
+                    requesterAddress,
```

Comment:
@fabriziovigevani here is what i mention on the comment, you need to pass to the renderEnty which fields are you going to show